### PR TITLE
Final adaptations for System Fetcher mTLS

### PR DESCRIPTION
**Pull request by danielgospodinow**
**Description**

Enable only mTLS communication in the System Fetcher component.

**Pull Request status**

- [x] Implementation
- [ ] *N/A* Unit tests
- [ ] *N/A* Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [ ] *N/A* Mocks are regenerated, using the automated script

### CODE REVIEW

**Review by PetarTodorovv — CHANGES_REQUESTED** [review]

**Comment by PetarTodorovv** [comment]
As part of https://github.com/kyma-incubator/compass/pull/2206 I have to make these changes as well so we can briefly discuss it and eventually close this PR?

File: chart/compass/templates/system-fetcher-job.yaml [hunk 1]
```diff
@@ -169,8 +164,10 @@ metadata:
   namespace: {{ .Release.Namespace }}
 data:
   {{ .Values.global.systemFetcher.secret.clientIdKey }}: {{ .Values.global.systemFetcher.oauth.client | b64enc | quote }}
-  {{ .Values.global.systemFetcher.secret.clientSecretKey }}: {{ .Values.global.systemFetcher.oauth.secret | b64enc | quote }}
   {{ .Values.global.systemFetcher.secret.oauthUrlKey }}: {{ printf "%s.%s" .Values.global.systemFetcher.oauth.tokenBaseUrl .Values.global.ingress.domainName | b64enc | quote }}
+  {{ if .Values.global.systemFetcher.secret.clientSecretKey }}
+  {{ .Values.global.systemFetcher.secret.clientSecretKey }}: {{ .Values.global.systemFetcher.oauth.secret | b64enc | quote }}
+  {{ end }}
```
[the remark this answers is not in the document]
Comment (danielgospodinow) [reply chart/compass/templates/system-fetcher-job.yaml:170]: I've left it only for backwards compatibility.

[the remark this answers is not in the document]
Comment (danielgospodinow) [reply chart/compass/templates/system-fetcher-job.yaml:170]: I removed it.

File: chart/compass/values.yaml [hunk 2]
```diff
@@ -525,7 +525,6 @@ global:
     secret:
       name: "compass-system-fetcher-secret"
       clientIdKey: client-id
-      clientSecretKey: client-secret
```
[the remark this answers is not in the document]
Comment (danielgospodinow) [reply chart/compass/values.yaml:528]: I'm not sure what has to be done additionally in the `config` repo.

File: installation/resources/installer-config-local.yaml.tpl [hunk 3]
```diff
@@ -46,14 +46,11 @@ data:
...
-  global.systemFetcher.oauth.secret: "client_secret"
-  global.systemFetcher.oauth.tokenBaseUrl: "compass-external-services-mock"
-  global.systemFetcher.oauth.tokenPath: "/secured/oauth/token"
+  global.systemFetcher.oauth.tokenBaseUrl: "compass-external-services-mock-sap-mtls"
+  global.systemFetcher.oauth.tokenPath: "/cert/token"
   global.systemFetcher.oauth.tokenEndpointProtocol: "https"
   global.systemFetcher.oauth.scopesClaim: "scopes"
   global.systemFetcher.oauth.tenantHeaderName: "x-zid"
-  global.systemFetcher.oauth.tokenRequestTimeout: "10s"
-  global.systemFetcher.oauth.skipSSLValidation: "true"
```
[the remark this answers is not in the document]
Comment (danielgospodinow) [reply installation/resources/installer-config-local.yaml.tpl:56]: Sure, I guess there can be more properties that can be removed. I just removed only those relevant to the task. I can take care of the remaining ones as well.

**Review by PetarTodorovv — APPROVED** [review]

**Pull request merged**